db/model/subject.js

```diff
@@ -225,9 +225,9 @@ module.exports = function subject(seq, dataTypes) {
                 batch.getSubjAspMapMembers(oldAbsPath)
               );
 
-              batch.removeSubjectTags(inst._previousDataValues);
+              batch.removeKeysForSubject(inst._previousDataValues);
               if (!tagsSetCreated) {
-                batch.setSubjectTags(inst);
+                batch.setupKeysForSubject(inst);
```

Comment:
Do we need these corresponding changes for aspect also?

Reply:
For aspects we are already calling setupKeysForAspect(), all I had to do was add setAspectExists() to that.